### PR TITLE
Help parsing list of objects

### PR DESCRIPTION
**Pull request by tiferrei**
Good evening,

Sorry to bother but I am a bit stuck on the final part of completing the Project GET requests (new ones were added on the latest version of GitLab), however I am stuck on parsing the JSON from the `Commit` class, into the `EventData` class, this is different from other cases as `EventData` would have a `commits` variable with _multiple_ `Commit` objects, being so a list of `Commit` objects, not just a single one.

The specific line is:
```swift
commits =  (json["commits"] as? [[String: AnyObject]]).map { Commit($0) } // TODO: Check class list parsing
```

Any help and tips are greatly appreciated.

Thank you,
tiferrei

PS: Please ignore extra edits from master, including Commit requests and Event requests, everything will be tidied up once this bump is fixed.

### CODE REVIEW

File: TanukiKit/Project.swift [hunk 1]
```diff
@@ -174,9 +239,75 @@ public enum VisibilityLevel: Int {
...
+        message = json["message"] as? String
+        userID = json["user_id"] as? Int
+        userName = json["user_name"] as? String
+        userEmail = json["user_email"] as? String
+        if let urlString = json["user_avater"] as? String, let urlFromString = URL(string: urlString) {
+            userAvatar = urlFromString
+        }
+        projectID = json["project_id"] as? Int
+        project = Project(json["project"] as? [String: AnyObject] ?? [:])
+        commits =  (json["commits"] as? [[String: AnyObject]]).map { Commit($0) } // TODO: Check class list parsing
```
Comment (tiferrei) [inline TanukiKit/Project.swift:276]: @pietbrauer: This is the specific line that is giving me trouble.

Thank you for your time.

Comment (pietbrauer) [reply TanukiKit/Project.swift:276]: Looks ok to me? What is giving you troubles?

Comment (tiferrei) [reply TanukiKit/Project.swift:276]: Well, on compilation it gives me: `Cannot convert value of type '[[String : AnyObject]]' to expected argument type '[String : Any]'`
But if I use only one pair of brackets it tells me: `Cannot convert value of type 'Commit' to closure result type '[Commit]'`

BTW: Happy New Year! 🎉

Comment (pietbrauer) [reply TanukiKit/Project.swift:276]: Happy New Year to you too!

You are just missing a `?` as you are mapping on an optional value:

```swift
commits =  (json["commits"] as? [[String: AnyObject]])?.map { Commit($0) } // TODO: Check class list parsing
```

Comment (tiferrei) [reply TanukiKit/Project.swift:276]: Awesome, that did it! Thanks a lot! 👍

**Comment by tiferrei** [comment]
Closing this one as fixed.

**Pull request closed without merging**